lib/iris/tests/unit/experimental/ugrid/test_MeshCoord.py

```diff
@@ -0,0 +1,394 @@
+    return kwargs
+
+
+def _create_test_meshcoord(**override_kwargs):
+    kwargs = _default_create_args()
+    # Apply requested overrides and additions.
+    kwargs.update(override_kwargs)
+    # Create and return the test coord.
+    result = MeshCoord(**kwargs)
+    return result
+
+
+class Test___init__(tests.IrisTest):
+    def setUp(self):
+        self.meshcoord = _create_test_meshcoord()
+
+    def test_basic(self):
+        kwargs = _default_create_args()
+        meshcoord = _create_test_meshcoord(**kwargs)
+        for key, val in kwargs.items():
+            self.assertEqual(getattr(meshcoord, key), val)
+        self.assertIsInstance(meshcoord, MeshCoord)
+        self.assertIsInstance(meshcoord, Coord)
+
+    def test_derived_properties(self):
+        # Check the derived properties of the meshcoord against the correct
+        # underlying mesh coordinate.
+        for axis in ("x", "y"):
+            meshcoord = _create_test_meshcoord(axis=axis)
+            # N.B.
+            node_x_coords = meshcoord.mesh.coord(include_nodes=True, axis=axis)
+            (node_x_coord,) = list(node_x_coords.values())
+            for key in node_x_coord.metadata._fields:
+                meshval = getattr(meshcoord, key)
+                if key == "var_name":
+                    # var_name is unused.
+                    self.assertIsNone(meshval)
+                else:
+                    # names, units and attributes are derived from the node coord.
+                    self.assertEqual(meshval, getattr(node_x_coord, key))
+
+    def test_fail_bad_mesh(self):
+        with self.assertRaisesRegex(ValueError, "must be a.*Mesh"):
+            _create_test_meshcoord(mesh=mock.sentinel.odd)
+
+    def test_valid_locations(self):
+        for loc in ("face", "edge", "node"):
+            meshcoord = _create_test_meshcoord(location=loc)
+            self.assertEqual(meshcoord.location, loc)
+
+    def test_fail_bad_location(self):
+        with self.assertRaisesRegex(ValueError, "not a valid Mesh location"):
+            _create_test_meshcoord(location="bad")
+
+    def test_fail_bad_axis(self):
+        with self.assertRaisesRegex(ValueError, "not a valid Mesh axis"):
+            _create_test_meshcoord(axis="q")
+
+
+class Test__readonly_properties(tests.IrisTest):
+    def setUp(self):
+        self.meshcoord = _create_test_meshcoord()
+
+    def test_fixed_metadata(self):
+        # Check that you cannot set any of these on an existing MeshCoord.
+        meshcoord = self.meshcoord
+        for prop in ("mesh", "location", "axis"):
+            with self.assertRaisesRegex(AttributeError, "can't set"):
+                setattr(meshcoord, prop, mock.sentinel.odd)
+
+    def test_coord_system(self):
+        # The property exists, =None, can set to None, can not set otherwise.
+        self.assertTrue(hasattr(self.meshcoord, "coord_system"))
+        self.assertIsNone(self.meshcoord.coord_system)
+        self.meshcoord.coord_system = None
```

Comment:
@pp-mo This is slightly confusing... you shouldn't be able to set `coord_system` to anything, it should be completely read-only.

Reply:
Please see notes  above.
I provide a setter, and allow it to write ONLY None.
Because the parent constructor requires to write this, unless we rewrite it somehow.  I really don't **_want_** to change the method by which this is set, since in all other cases it is part of the metadata so the mechanism is provided 'automatically'.